cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -379,6 +457,109 @@ writeLedgerState mOutFile qState@(SerialisedLedgerState serLedgerState) =
    decodeLedgerState (SerialisedLedgerState (Serialised ls)) =
      first (const ls) (decodeFull ls)
 
+writeStakeSnapshot :: forall era ledgerera.
+                      ShelleyLedgerEra era ~ ledgerera
+                   => Era.Era ledgerera
+                   => FromCBOR (LedgerState era)
+                   => Hash StakePoolKey
+                   -> SerialisedLedgerState era
+                   -> ExceptT ShelleyQueryCmdError IO ()
+
+writeStakeSnapshot poolId qState =
+  case decodeLedgerState qState of
+    Left bs -> firstExceptT ShelleyQueryCmdHelpersError $ pPrintCBOR bs
+    Right ledgerState ->
+      if isNothing maybehk then
+        left $ ShelleyQueryCmdPoolIdError poolId
+      else
+        liftIO . LBS.putStrLn $  encodePretty $ Stakes {markpool = markStake, setpool = setStake, gopool = goStake, marktot = markTotal, settot = setTotal, gotot = goTotal}
+      where
+        -- Ledger State
+        (LedgerState snapshot) = ledgerState
+
+        -- The three stake snapshots, obtained from the ledger state
+        (SnapShots markS setS goS _) = esSnapshots $ nesEs snapshot
+
+        -- Calculate the three pool and active stake values for the given pool
+        markStake =  getPoolStake hk markS
+        setStake = getPoolStake hk setS
+        goStake = getPoolStake hk goS
+
+        markTotal = getAllStake markS
+        setTotal = getAllStake setS
+        goTotal = getAllStake goS
+
+        -- Convert the hash string into a KeyHash for use by the ledger
+        maybehk = hashFromStringAsHex $ filter (/= '"') $ show poolId
+        hk = KeyHash $ fromJust maybehk
+ where
+   decodeLedgerState
+     :: SerialisedLedgerState era
+     -> Either LBS.ByteString (LedgerState era)
+   decodeLedgerState (SerialisedLedgerState (Serialised ls)) =
+     first (const ls) (decodeFull ls)
+
+   -- Sum all the stake that is held by the pool
+   getPoolStake :: KeyHash Shelley.Spec.Ledger.Keys.StakePool crypto
+                -> SnapShot crypto
+                -> Integer
+   getPoolStake hk ss = pStake
+      where
```

Comment:
We don't need to nest these `where` clauses. We can write it as one `where` clause:
```
      where
        Coin pStake = fold s
        (Stake s) = poolStake hk (_delegations ss) (_stake ss)
  ```